packages/ragbits-core/src/ragbits/core/prompt/prompt.py

```diff
@@ -74,6 +75,16 @@ def _render_template(cls, template: Template, input_data: Optional[InputT]) -> s
             context = input_data.model_dump(serialize_as_any=True)
         return template.render(**context)
 
+    def _get_images_from_input_data(self, input_data: Optional[InputT]) -> list:
+        images = []
+        if isinstance(input_data, BaseModel):
+            context = input_data.model_dump(serialize_as_any=True)
```

Comment:
Why do you serialize the input data model? There is no need for this - you can inspect the object directly, without loosing the precise typing and metadata, and without spending time on serialization.

Reply:
has been addressed